deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -756,3 +838,40 @@ def __init__(self, tf_class, **kwargs):
 
   def __call__(self):
     return self.tf_class(**self.kwargs)
+
+
+class Submodel(object):
```

Comment:
Should a submodel be a top level class or perhaps inherit from one of our existing abstract classes?

Reply:
Which one did you have in mind?

Reply:
It's actually a little tricky now that I think about it. It feels like `Submodel` and `Model` should be related somehow but maybe that just doesn't work out under Python's current inheritance semantics.

Reply:
It's nothing to do with Python.  They're just different concepts.  Remember I suggested "submodel" should really be called "training objective" instead?  That's what it really is.  You preferred the name "submodel", so that's what I used, but remember that it *isn't* a model.  It's a training objective.

Reply:
Let's rename it in that case :). I was envisioning something more related to a model object itself.